R/io_ipc.R

```diff
@@ -32,7 +29,6 @@ pl_scan_ipc = function(
     source,
     ...,
     n_rows = NULL,
-    memory_map = TRUE,
```

Comment:
Is this change needed?

Reply:
I think so because of https://github.com/pola-rs/r-polars/pull/1104/commits/1eb42452e6740e67b58352ab1849be640fba4f02

Reply:
Oh, I see. Perhaps it was only removed from `.pr$DataFrame$from_raw_ipc`, so there should be no need to remove it from anywhere else.
I suppose that this is still valid when reading from files.

Reply:
I'm confused, I don't know if `memory_map` should be removed from other places functions and from all tests where it is set. I also don't really understand what this argument does, so can you either deal with this or let me know if it's fine to remove it from the `scan/read_ipc` functions?

Reply:
This only makes sense when reading from an Arrow file. My understanding is that it is an option to determine whether or not to keep a reference to the file without copying the data, and disabling it will degrade performance.

``` r
polars::pl$DataFrame(a = 1)$write_ipc("test.arrow")

df <- polars::pl$read_ipc("test.arrow")
df2 <- polars::pl$read_ipc("test.arrow", memory_map = FALSE)

df
#> shape: (1, 1)
#> ┌─────┐
#> │ a   │
#> │ --- │
#> │ f64 │
#> ╞═════╡
#> │ 1.0 │
#> └─────┘
```

``` r
df2
#> shape: (1, 1)
#> ┌─────┐
#> │ a   │
#> │ --- │
#> │ f64 │
#> ╞═════╡
#> │ 1.0 │
#> └─────┘
```

``` r

polars::pl$DataFrame(a = 2)$write_ipc("test.arrow")

df
#> shape: (1, 1)
#> ┌─────┐
#> │ a   │
#> │ --- │
#> │ f64 │
#> ╞═════╡
#> │ 2.0 │
#> └─────┘
```

``` r
df2
#> shape: (1, 1)
#> ┌─────┐
#> │ a   │
#> │ --- │
#> │ f64 │
#> ╞═════╡
#> │ 1.0 │
#> └─────┘
```

<sup>Created on 2024-05-26 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>